lucene/core/src/java/org/apache/lucene/index/NormValuesWriter.java

```diff
@@ -76,7 +76,8 @@ public void flush(SegmentWriteState state, Sorter.DocMap sortMap, NormsConsumer
           NumericDocValuesWriter.sortDocValues(
               state.segmentInfo.maxDoc(),
               sortMap,
-              new BufferedNorms(values, docsWithField.iterator()));
+              new BufferedNorms(values, docsWithField.iterator()),
+              docsWithField.dense() && sortMap.size() == docsWithField.cardinality());
```

Comment:
Only testing `dense()` should be correct right?

```suggestion
              docsWithField.dense());
```

Reply:
no,DocsWithFieldSet will update it's bitset when a doc with field really added only, in sparse, it will not call call DocsWithFieldSet#add, so if the first 64 doc with field has added, and then some doc added without this filed, the docsWithField.dense() will return true, i think we can remove `docsWithField.dense()`, use  `sortMap.size() == docsWithField.cardinality()` only for dense case, the sortMap.size() will return the number of documents for the LeafReader, what do you think?

Reply:
you're right! thanks for explaining

Reply:
i have deleted docsWithField.dense()